CONTRIBUTING.md

```diff
@@ -7,8 +7,103 @@ We use the [black](https://pypi.org/project/black/) python linter. You can have
 We use Google Docstrings. Please refer to [this example](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).
 
 ## Typing
-We also use static typing for our function arguments/variables for better code readability. Our continuously integrated testing uses `pytest --mypy .` to run tests, which runs static type checking. 
+We also use static typing for our function arguments/variables for better code readability. We have a github action that runs `mypy .`, which runs similar to `pytest .` to check for valid static typing. You can refer to [mypy documentation](https://mypy.readthedocs.io/en/stable/) for more information.
 
-The `--mypy` flag uses the [pytest-mypy](https://pypi.org/project/pytest-mypy/) pytest plugin.
+## Testing
+We use [pytest](https://docs.pytest.org/en/6.2.x/) for our tests. In order to make it easier, we also have a set of custom options defined in [conftest.py](conftest.py).
 
 
+### Prerequisites
+- Understand how to write [pytest](https://docs.pytest.org/en/6.2.x/) tests.
+- Understand what a [pytest fixture](https://docs.pytest.org/en/6.2.x/fixture.html) is.
+- Understand what [pytest parametrizations](https://docs.pytest.org/en/6.2.x/parametrize.html) are.
+
+
+### Options
+An option is provided as such: `pytest . --some-option`.
+
+**Note: When running `pytest .`, none of these options are provided. The only options provided by default are `-s` and `--benchmark-skip`.**
+
+- `--memory-skip`: Skip all tests that use `MemoryProvider`. Either as a parametrization (in the `storage` fixture) or standalone as the `memory_storage` fixture.
+- `--local`: Run all tests that use `LocalProvider`. Either as a parametrization (in the `storage` fixture) or standalone as the `local_storage` fixture.
+- `--s3`: Run all tests that use `S3Provider`. Either as a parametrization (in the `storage` fixture) or standalone as the `s3_storage` fixture.
+- `--cache-chains`: Run all tests that use a chain of `StorageProvider`s. Only enabled (via the options above) `StorageProvider`s will be used for composing these cache chains. Cache chains will always be provided as a parametrization of the `storage` fixture.
+- `--cache-chains-only`: Force enables `--cache-chains`, but all parametrizations/fixtures that contain non-cache chain `StorageProvider`s will be skipped.
+- `--s3-path`: Url to s3 bucket with optional key to be used for the `S3Provider`. Example: s3://bucket_name/key/to/tests/. The default value can be found in [hub.constants](hub/constants.py).
+- `--keep-storage`: By default, all `StorageProvider`s created via fixtures/parametrizations will have their data wiped (unless it is an `S3Provider`). This option will keep the storage data.
```

Comment:
Why don't we wipe S3Providers?

Reply:
This might be outdated. Calling clear on S3 was slower earlier but I deployed a fix a few days back that made it much faster after which we removed this clause of not clearing S3, right @McCrearyD?

Reply:
right right